### PR TITLE
MWPW-153580: Add Opt-In Feature for CaaS Badge Display

### PR DESCRIPTION
**Pull request by sanrai**
### TLDR
We added a checkbox in CaaS tool configurator to enable a new badging feature. 

Milo only contains the logic for toggling a checkbox true or false. 

All the actual implementation is done on the CaaS side.

Note: This feature only affects an authoring tool. The LH scores tested by the GH actions are irrelevant in this context.

### Description

This pull request introduces a new feature for Milo, allowing authors to opt-in to display badges on their CaaS cards. Previously, Milo was incorrectly sending the `primaryTag` information for each card. Specifically, for Milo (using the postXDM service), badge label information was being sent incorrectly as an ID _(in the wrong structure),_ missing the tag's title and localized values.

To resolve this, we will use static tag data inside Chimera IO to look up the correct tag title based on the ID passed on by Milo. This approach is much easier than fixing the postXDM service and performing a database migration in OpenSearch.

By implementing this feature, Milo authors can now choose to display badges by enabling a checkbox.

Resolves: [MWPW-153580](https://jira.corp.adobe.com/browse/MWPW-153580)

### Additional Context
For cards from Dexter, badges should always display by default. For cards from Milo, authors may not expect badges unless they opt in. This prevents unintended badge displays for Milo authors who did not expect them _(by accidentally authoring `primaryTag` fields without realizing they show badges)_

#### Handling Different Content Origins:

We have consumers from Dexter and Milo. 

- For content from Dexter, badges should always show (default behavior).
- For content from Milo, badges should only show if the author has opted in.

#### Code Changes:
- Added a checkbox in the UI to allow Milo authors to opt-in for showing badges.

#### Backend Code:
```js
const miloBadgeLabel = primaryTag.id ? localizeTag(primaryTag.id, language, country) : '';
...
{ label: { description: (primaryTag.title || miloBadgeLabel) } } : { label: { description: '' } }),
````

#### CaaS UI Code:
```js
const showCardBadges = getConfig('collection', 'showCardBadges');
const fromDexter = origin === 'Dexter';
const showBadge = (isOneHalf || isThreeFourths || isFull) && (fromDexter || showCardBadges);
```

#### Milo Code:
```js
showCardBadges: state.showCardBadges,
...
<${Input} label="Enable showing card badges (by default hidden)" prop="showCardBadges" type="checkbox" />
```

### Testing
- Verify that the opt-in checkbox is available for Milo authors.
- Verify that badges are displayed correctly for Dexter content by default.
- Verify that badges are displayed for Milo content only when opted in by the author.
- Ensure that no badges are shown for Milo content if the author has not opted in.

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/caas?martech=off
image

- After: https://MWPW-153580--milo--adobecom.hlx.page/tools/caas?martech=off
image

By merging this pull request, we will ensure that authors have control over badge visibility based on the content origin and their preferences.

### CODE REVIEW

**Review by overmyheadandbody — COMMENTED** [review]
I see some failing checks that might deserve some attention. I also see this is marked as `do not merge`. Is it a draft? Will you be adding additional logic?

**Comment by sanrai** [comment]
Hi @overmyheadandbody,

I was in the process of raising the PR last night just before the end of the work day but didn’t have enough time to finish it up until today.

The PR is now complete with all the documentation written and tests updated. I have also addressed the failing checks and everything should be in order now :) 
 
> I see some failing checks that might deserve some attention. I also see this is marked as do not merge. Is it a draft? Will you be adding additional logic?

**Review by mokimo — APPROVED** [review]

**Review by overmyheadandbody — APPROVED** [review]

**Review by cmiqueo — APPROVED** [review]

**Review by arshadparwaiz — APPROVED** [review]
